Getting connection error when trying to execute flow using the get task

### Describe the issue

I was trying to fetch all the pods from my local minikube cluster using the kubernetes plugin. Below is the flow(its not the entire but its already failing at the first task so i'm not pasting all the code)

```yaml
id: k8s-ai-incident-monitor
namespace: devops.monitoring

description: |
  Kubernetes AI-powered incident monitoring and alerting system.
  Monitors cluster health, analyzes issues with AI, and sends intelligent notifications.

variables:
  kubernetes:
    namespace: default
  slack:
    channel: "#k8s-alerts"
    critical_channel: "#critical-incidents"

triggers:
  - id: schedule-monitor
    type: io.kestra.plugin.core.trigger.Schedule
    cron: "*/5 * * * *" # Every 5 minutes
  
  - id: webhook-alert
    type: io.kestra.plugin.core.trigger.Webhook
    key: 4wjtkzwVGBM9yKnjm3yv8r
    conditions:
      - type: io.kestra.plugin.core.condition.ExecutionStatus
        in:
          - FAILED
          - WARNING

tasks:
  # 1. Collect Kubernetes Cluster Data
  - id: collect-pods
    type: io.kestra.plugin.kubernetes.kubectl.Get
    namespace: default
    resourceType: pods
    fetchType: FETCH
    outputFiles: 
      - pods.json
```


I've pasted the error logs below:

```
2025-06-19 18:06:56.335Fetching all resources of kind 'pods' in namespace 'default'
2025-06-19 18:07:15.553kubernetes.default.svc
kubernetes.default.svc
2025-06-19 18:07:15.553Kubernetes API error while fetching kind 'pods' in namespace 'default': Operation: [list]  for kind: [pods]  with name: [null]  in namespace: [default]  failed.
2025-06-19 18:07:15.553io.fabric8.kubernetes.client.KubernetesClientException: Operation: [list]  for kind: [pods]  with name: [null]  in namespace: [default]  failed.
	at io.fabric8.kubernetes.client.KubernetesClientException.launderThrowable(KubernetesClientException.java:139)
	at io.fabric8.kubernetes.client.dsl.internal.BaseOperation.list(BaseOperation.java:453)
	at io.fabric8.kubernetes.client.dsl.internal.BaseOperation.list(BaseOperation.java:419)
	at io.fabric8.kubernetes.client.dsl.internal.BaseOperation.list(BaseOperation.java:98)
	at io.kestra.plugin.kubernetes.kubectl.Get.run(Get.java:174)
	at io.kestra.plugin.kubernetes.kubectl.Get.run(Get.java:35)
	at io.kestra.core.runners.WorkerTaskCallable.doCall(WorkerTaskCallable.java:79)
	at io.kestra.core.runners.AbstractWorkerCallable.call(AbstractWorkerCallable.java:63)
	at io.kestra.core.runners.WorkerSecurityService.callInSecurityContext(WorkerSecurityService.java:10)
	at io.kestra.core.runners.Worker.lambda$callJob$19(Worker.java:853)
	at io.kestra.core.trace.NoopTracer.inCurrentContext(NoopTracer.java:15)
	at io.kestra.core.runners.Worker.callJob(Worker.java:849)
	at io.kestra.core.runners.Worker.runAttempt(Worker.java:807)
	at io.kestra.core.runners.Worker.run(Worker.java:654)
	at io.kestra.core.runners.Worker.handleTask(Worker.java:353)
	at io.kestra.core.runners.Worker.lambda$run$8(Worker.java:276)
	at io.micrometer.core.instrument.internal.TimedRunnable.run(TimedRunnable.java:49)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.io.IOException: kubernetes.default.svc
	at io.fabric8.kubernetes.client.dsl.internal.OperationSupport.waitForResult(OperationSupport.java:504)
	at io.fabric8.kubernetes.client.dsl.internal.BaseOperation.list(BaseOperation.java:451)
	... 18 more
Caused by: java.net.UnknownHostException: kubernetes.default.svc
	at java.base/java.net.InetAddress$CachedLookup.get(Unknown Source)
	at java.base/java.net.InetAddress.getAllByName0(Unknown Source)
	at java.base/java.net.InetAddress.getAllByName(Unknown Source)
	at java.base/java.net.InetAddress.getByName(Unknown Source)
	at io.netty.util.internal.SocketUtils$8.run(SocketUtils.java:156)
	at io.netty.util.internal.SocketUtils$8.run(SocketUtils.java:153)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at io.netty.util.internal.SocketUtils.addressByName(SocketUtils.java:153)
	at io.netty.resolver.DefaultNameResolver.doResolve(DefaultNameResolver.java:41)
	at io.netty.resolver.SimpleNameResolver.resolve(SimpleNameResolver.java:61)
	at io.netty.resolver.SimpleNameResolver.resolve(SimpleNameResolver.java:53)
	at io.netty.resolver.InetSocketAddressResolver.doResolve(InetSocketAddressResolver.java:55)
	at io.netty.resolver.InetSocketAddressResolver.doResolve(InetSocketAddressResolver.java:31)
	at io.netty.resolver.AbstractAddressResolver.resolve(AbstractAddressResolver.java:106)
	at io.netty.bootstrap.Bootstrap.doResolveAndConnect0(Bootstrap.java:220)
	at io.netty.bootstrap.Bootstrap.doResolveAndConnect(Bootstrap.java:171)
	at io.netty.bootstrap.Bootstrap.connect(Bootstrap.java:148)
	at io.vertx.core.net.impl.ChannelProvider.handleConnect(ChannelProvider.java:152)
	at io.vertx.core.net.impl.ChannelProvider.connect(ChannelProvider.java:103)
	at io.vertx.core.net.impl.ChannelProvider.connect(ChannelProvider.java:89)
	at io.vertx.core.net.impl.NetClientImpl.connectInternal2(NetClientImpl.java:308)
	at io.vertx.core.net.impl.NetClientImpl.lambda$connectInternal2$7(NetClientImpl.java:328)
	at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:173)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:166)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:569)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:998)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	... 1 more
```


I was using:

```
docker run --pull=always --rm -it -p 8080:8080 \
>   -e SECRET_GOOGLE_API_KEY=........\
>   --user=root \
>   -v /var/run/docker.sock:/var/run/docker.sock \
>   -v /tmp:/tmp \
>   -v /tmp/ca.crt:/root/.kube/ca.crt:ro \
>   kestra/kestra:latest server local
```

This command to start Kestra. First I didn't passed the connection prop as it should get the default config automatically but later I tried to connect using the 

```
connection:
  masterUrl:   # which i got from kubectl config view --raw and used the server value 
  caCertFile: "/root/.kube/ca.crt"
  oauthToken:    # i got this using kubectl create token kestra-sa --duration=24h
```

This also gave error saying refused to connect to the passed masterUrl . maybe I used the wrong one but its the correct one ig.

### Environment

- Kestra Version: latest image

--- a/.gitignore
+++ b/.gitignore
@@ -14,3 +14,5 @@ out/
 .classpath
 .attach*
 src/test/resources/application-test.yml
+kind
+kubectl

--- a/src/main/java/io/kestra/plugin/kubernetes/kubectl/Apply.java
+++ b/src/main/java/io/kestra/plugin/kubernetes/kubectl/Apply.java
@@ -57,6 +57,63 @@ import java.util.List;
                     namespace: default
                     spec: "{{ read('deployment.yaml') }}"
                 """
+        ),
+        @Example(
+            title = "Apply a Kubernetes custom resource definition, using YAML.",
+            full = true,
+            code = """
+                id: k8s
+                namespace: company.name
+
+                tasks:
+                  - id: apply
+                    type: io.kestra.plugin.kubernetes.kubectl.Apply
+                    namespace: default
+                    spec: |-
+                      apiVersion: apiextensions.k8s.io/v1
+                      kind: CustomResourceDefinition
+                      metadata:
+                        name: shirts.stable.example.com
+                      spec:
+                        group: stable.example.com
+                        scope: Namespaced
+                        names:
+                          plural: shirts
+                          singular: shirt
+                          kind: Shirt
+                        versions:
+                        - name: v1
+                          served: true
+                          storage: true
+                          schema:
+                            openAPIV3Schema:
+                              type: object
+                              properties:
+                                apiVersion:
+                                  type: string
+                                kind:
+                                  type: string
+                                metadata:
+                                  type: object
+                                spec:
+                                  type: object
+                                  x-kubernetes-preserve-unknown-fields: true # Allows any fields in spec
+                                  properties:
+                                    # You should define your actual Shirt properties here later
+                                    # For example:
+                                    # color:
+                                    #   type: string
+                                    # size:
+                                    #   type: string
+                                    #   enum: ["S", "M", "L", "XL"]
+                                status:
+                                  type: object
+                                  x-kubernetes-preserve-unknown-fields: true # Allows any fields in status
+                                  properties:
+                                    # Define your status properties here
+                                    # message:
+                                    #   type: string
+                """
         )
     }
 )

--- a/src/main/java/io/kestra/plugin/kubernetes/kubectl/Delete.java
+++ b/src/main/java/io/kestra/plugin/kubernetes/kubectl/Delete.java
@@ -69,6 +69,16 @@ public class Delete extends AbstractPod implements RunnableTask<VoidOutput> {
     @NotNull
     private Property<List<String>> resourcesNames;
 
+    @Schema(
+        title = "The Kubernetes resource apiGroup"
+    )
+    private Property<String> apiGroup;
+
+    @Schema(
+        title = "The Kubernetes resource apiVersion"
+    )
+    private Property<String> apiVersion;
+
     @Override
     public VoidOutput run(RunContext runContext) throws Exception {
 
@@ -79,12 +89,14 @@ public class Delete extends AbstractPod implements RunnableTask<VoidOutput> {
             var renderedKind = runContext.render(this.resourceType).as(String.class)
                 .orElseThrow(() -> new IllegalArgumentException("resourceType must be provided and rendered."));
             var renderedResourcesNames = runContext.render(this.resourcesNames).asList(String.class);
+            var renderedApiGroup = runContext.render(this.apiGroup).as(String.class).orElse("");
+            var renderedApiVersion = runContext.render(this.apiVersion).as(String.class).orElse("v1");
 
             runContext.logger().debug("Deleting resource(s) '{}' of kind '{}' in namespace '{}'", renderedResourcesNames, renderedKind, renderedNamespace);
 
             var resourceDefinitionContext = new ResourceDefinitionContext.Builder()
-                .withGroup("apps")
-                .withVersion("v1")
+                .withGroup(renderedApiGroup)
+                .withVersion(renderedApiVersion)
                 .withKind(renderedKind)
                 .withNamespaced(true) // Assuming resources are namespaced as we take namespace input
                 .build();

--- a/src/main/java/io/kestra/plugin/kubernetes/kubectl/Get.java
+++ b/src/main/java/io/kestra/plugin/kubernetes/kubectl/Get.java
@@ -152,7 +152,7 @@ public class Get extends AbstractPod implements RunnableTask<Get.Output> {
         var renderedResourceType = runContext.render(this.resourceType).as(String.class)
             .orElseThrow(() -> new IllegalArgumentException("resourceType must be provided and rendered."));
         var renderedResourcesNames = runContext.render(this.resourcesNames).asList(String.class);
-        var renderedApiGroup = runContext.render(this.apiGroup).as(String.class).orElse("apps");
+        var renderedApiGroup = runContext.render(this.apiGroup).as(String.class).orElse("");
         var renderedApiVersion = runContext.render(this.apiVersion).as(String.class).orElse("v1");
         var renderedFetchType = runContext.render(this.fetchType).as(FetchType.class).orElse(NONE);
 
@@ -161,6 +161,7 @@ public class Get extends AbstractPod implements RunnableTask<Get.Output> {
         try (var client = PodService.client(runContext, this.getConnection())) {
 
             var resourceDefinitionContext = new ResourceDefinitionContext.Builder()
+                // See: https://kubernetes.io/docs/reference/using-api/api-concepts/#resource-uris
                 .withGroup(renderedApiGroup)
                 .withVersion(renderedApiVersion)
                 .withKind(renderedResourceType)

--- a/src/test/java/io/kestra/plugin/kubernetes/kubectl/ApplyTest.java
+++ b/src/test/java/io/kestra/plugin/kubernetes/kubectl/ApplyTest.java
@@ -2,7 +2,6 @@ package io.kestra.plugin.kubernetes.kubectl;
 
 import io.kestra.core.junit.annotations.KestraTest;
 import io.kestra.core.models.property.Property;
-import io.kestra.core.runners.RunContext;
 import io.kestra.core.runners.RunContextFactory;
 import jakarta.inject.Inject;
 import lombok.extern.slf4j.Slf4j;

--- a/src/test/java/io/kestra/plugin/kubernetes/kubectl/DeleteTest.java
+++ b/src/test/java/io/kestra/plugin/kubernetes/kubectl/DeleteTest.java
@@ -67,6 +67,8 @@ public class DeleteTest {
             .namespace(Property.ofValue(DEFAULT_NAMESPACE))
             .resourceType(Property.ofValue("deployments"))
             .resourcesNames(Property.ofValue(List.of("my-deployment")))
+            .apiGroup(Property.ofValue("apps"))
+            .apiVersion(Property.ofValue("v1"))
             .fetchType(Property.ofValue(FetchType.FETCH_ONE))
             .build();
 
@@ -82,6 +84,8 @@ public class DeleteTest {
             .namespace(Property.ofValue(DEFAULT_NAMESPACE))
             .resourceType(Property.ofValue("deployments"))
             .resourcesNames(Property.ofValue(List.of("my-deployment")))
+            .apiGroup(Property.ofValue("apps"))
+            .apiVersion(Property.ofValue("v1"))
             .build();
 
         // Then
@@ -147,6 +151,8 @@ public class DeleteTest {
             .type(Get.class.getName())
             .namespace(Property.ofValue(DEFAULT_NAMESPACE))
             .resourceType(Property.ofValue("deployments"))
+            .apiGroup(Property.ofValue("apps"))
+            .apiVersion(Property.ofValue("v1"))
             .fetchType(Property.ofValue(FetchType.FETCH))
             .build();
 
@@ -162,6 +168,8 @@ public class DeleteTest {
             .namespace(Property.ofValue(DEFAULT_NAMESPACE))
             .resourceType(Property.ofValue("deployments"))
             .resourcesNames(Property.ofValue(deploymentsNames))
+            .apiGroup(Property.ofValue("apps"))
+            .apiVersion(Property.ofValue("v1"))
             .build();
 
         // Then

--- a/src/test/java/io/kestra/plugin/kubernetes/kubectl/GetTest.java
+++ b/src/test/java/io/kestra/plugin/kubernetes/kubectl/GetTest.java
@@ -74,6 +74,8 @@ public class GetTest {
             .namespace(Property.ofValue(DEFAULT_NAMESPACE))
             .resourceType(Property.ofValue("deployments"))
             .resourcesNames(Property.ofValue(List.of(expectedDeploymentName)))
+            .apiGroup(Property.ofValue("apps"))
+            .apiVersion(Property.ofValue("v1"))
             .fetchType(Property.ofValue(FetchType.FETCH_ONE))
             .build();
 
@@ -138,6 +140,8 @@ public class GetTest {
             .type(Get.class.getName())
             .namespace(Property.ofValue(DEFAULT_NAMESPACE))
             .resourceType(Property.ofValue("deployments"))
+            .apiGroup(Property.ofValue("apps"))
+            .apiVersion(Property.ofValue("v1"))
             .fetchType(Property.ofValue(FetchType.FETCH))
             .build();
 
@@ -230,9 +234,9 @@ public class GetTest {
             .type(Get.class.getName())
             .namespace(Property.ofValue(DEFAULT_NAMESPACE))
             .resourceType(Property.ofValue(shirtKind))
-            .fetchType(Property.ofValue(FetchType.FETCH_ONE))
             .apiGroup(Property.ofValue(apiGroup))
             .apiVersion(Property.ofValue(apiVersion))
+            .fetchType(Property.ofValue(FetchType.FETCH_ONE))
             .build();
 
 

